go/libraries/doltcore/sqle/tables.go

```diff
@@ -96,6 +99,55 @@ func (t *DoltTable) PartitionRows(ctx *sql.Context, _ sql.Partition) (sql.RowIte
 	return newRowIterator(t, ctx)
 }
 
+// Insert adds the given row to the table and updates the database root.
+func (t *DoltTable) Insert(ctx *sql.Context, sqlRow sql.Row) error {
+	dRow, err := SqlRowToDoltRow(t.table.Format(), sqlRow, t.sch)
+	if err != nil {
+		return  err
+	}
+
+	taggedVals := row.TaggedValues{}
+	for _, pkTag := range t.sch.GetPKCols().Tags {
+		dCol, ok := dRow.GetColVal(pkTag)
+		if !ok {
+			return errors.New("primary key missing from values")
+		}
+		dColVal, err := dCol.Value(ctx)
```

Comment:
No need to assemble separate tagged values here. You can use drow.NomsMapKey to get the tuples you need for GetRowByPKVals

Reply:
Actually it's GetRowByPKTuple, and you will need to cast the result of NomsMapKey to call it. But it's better than doing additional IO to assemble the noms key.

Reply:
`GetRowByPKTuple` doesn't seem to exist anywhere in the codebase? I just did a full directory scan for that function and couldn't find anything.

Reply:
Nvm, the function doesn't exist but I looked up how `NomsMapKey` was used and figured it out from there.